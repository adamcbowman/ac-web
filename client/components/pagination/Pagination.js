import React, { createElement, PureComponent } from 'react'
import PropTypes from 'prop-types'
import CSSModules from 'react-css-modules'
import styles from './Pagination.css'
import Segment, { Disabled } from './Segment'
import range from 'lodash/range'
import pagination from 'utils/pagination'
import noop from 'lodash/noop'

@CSSModules(styles)
export default class Pagination extends PureComponent {
    static propTypes = {
        total: PropTypes.number.isRequired,
        active: PropTypes.number.isRequired,
        onChange: PropTypes.func.isRequired,
    }
    static defaultProps = {
        total: 0,
        active: 1,
        onChange: noop,
    }
    get segments() {
        const { active, total } = this.props

        if (total <= 10) {
            return range(1, total + 1)
        }

        return pagination(active, total, 3, null)
    }
    createSegment(page, index) {
        const { onChange, active } = this.props

        if (typeof page === 'number') {
            return createElement(Segment, {
                key: index,
                page,
                onActivate: onChange,
                isActive: active === page,
            })
        } else {
            return <Disabled key={index}>â€¦</Disabled>
        }
    }
    render() {
        if (this.props.total < 2) {
            return null
        }

        return (
            <div styleName="Container">
                {this.segments.map(this.createSegment, this)}
            </div>
        )
    }
}
