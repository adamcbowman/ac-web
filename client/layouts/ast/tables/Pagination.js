import React, { memo } from 'react'
import PropTypes from 'prop-types'
import Base from 'components/pagination'

Pagination.propTypes = {
    page: PropTypes.number.isRequired,
    pageSize: PropTypes.number,
    count: PropTypes.number.isRequired,
    onChange: PropTypes.func.isRequired,
}

function Pagination({ page, count, pageSize = 25, onChange }) {
    const total = Math.ceil(count / pageSize)

    return <Base total={total} active={page} onChange={onChange} />
}

export default memo(Pagination)
