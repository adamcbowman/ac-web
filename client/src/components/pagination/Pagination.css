@import "../../styles/colors.css";

.Container {
    display: flex;
    align-items: center;
    background-color: var(--gray-lighter);
    height: 80px;
    position: relative;
}


.Navigation {

}

.Description {
    margin: 5px;
}

.Hint {
    color: var(--brand-primary);
    line-height: 2em;
}

.Segment {
    flex: 1 50%;
    display: flex;
}

.Left {
    composes: Segment;
    justify-content: flex-start;
    /*border-right: white solid 2px;*/

    & > .Navigation {
        order: 1;
    }

    & > .Description {
        order: 2;
    }
}

.Right {
    composes: Segment;
    justify-content: flex-end;
    /*border-left: white solid 2px;*/

    & > .Description {
        order: 1;
    }

    & > .Navigation {
        order: 2;
    }

    & .Hint {
        text-align: right;
    }

}

.Left--hidden {
    composes: Left;
    visibility: hidden;
}
.Right--hidden {
    composes: Right;
    visibility: hidden;
}

.Center {
    padding: 10px;
}
