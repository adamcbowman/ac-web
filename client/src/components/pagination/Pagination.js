import React, {PropTypes, createElement} from 'react'
import CSSModules from 'react-css-modules'
import styles from './Pagination.css'
import Segment, {First, Previous, Next, Last} from './Segment'

Pagination.propTypes = {
    location: PropTypes.object.isRequired,
    total: PropTypes.number.isRequired,
    first: PropTypes.bool,
    previous: PropTypes.bool,
    next: PropTypes.bool,
    last: PropTypes.bool,
    max: PropTypes.number,
}

function toLocation(page, location) {
    return {
        ...location,
        query: {
            ...location.query,
            page,
        }
    }
}

function createSegment(page, location) {
    return createElement(Segment, {
        location: toLocation(page, location)
    }, page)
}

function Pagination({
    location = {},
    total = 0,
    first = false,
    previous = false,
    next = false,
    last = false,
    max = 10,
}) {
    if (total < 2) {
        return null
    }

    const list = Array(Math.min(total, max)).fill(null)

    return (
        <div styleName='Container'>
            {first && <First />}
            {previous && <Previous />}
            {list.map((value, index) => createSegment(index + 1, location))}
            {next && <Next />}
            {last && <Last />}
        </div>
    )
}

export default CSSModules(Pagination, styles)
