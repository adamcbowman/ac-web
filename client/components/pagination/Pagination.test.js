import React from 'react'
import Renderer from 'react-test-renderer'
import Pagination from './'

test('pagination component', () => {
    const pagination = Renderer.create(<Pagination total={10} active={5} />)

    expect(pagination).toMatchSnapshot()
})
