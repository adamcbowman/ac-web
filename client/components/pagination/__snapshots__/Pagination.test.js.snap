// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pagination component 1`] = `
<div
  className="Container"
>
  <a
    className="Segment"
    href="#"
    onClick={[Function]}
  >
    1
  </a>
  <a
    className="Segment"
    href="#"
    onClick={[Function]}
  >
    2
  </a>
  <a
    className="Segment"
    href="#"
    onClick={[Function]}
  >
    3
  </a>
  <a
    className="Segment"
    href="#"
    onClick={[Function]}
  >
    4
  </a>
  <a
    className="Segment--Active"
    href="#"
    onClick={[Function]}
  >
    5
  </a>
  <a
    className="Segment"
    href="#"
    onClick={[Function]}
  >
    6
  </a>
  <a
    className="Segment"
    href="#"
    onClick={[Function]}
  >
    7
  </a>
  <a
    className="Segment"
    href="#"
    onClick={[Function]}
  >
    8
  </a>
  <a
    className="Segment"
    href="#"
    onClick={[Function]}
  >
    9
  </a>
  <a
    className="Segment"
    href="#"
    onClick={[Function]}
  >
    10
  </a>
</div>
`;
